Simplify the voting process for developers (#339)
trilitech/tezos-developer-docs: docs/architecture/governance.md

---
title: Governance and self-amendment
last_update:
  date: 27 February 2024
---

Tezos incorporates a built-in, on-chain mechanism for proposing, selecting, testing, and activating protocol upgrades without the need to hard fork.

> This mechanism makes Tezos a self-amending blockchain and allows any user to propose changes to the [economic protocol](./architecture#the-tezos-self-amending-protocol), which defines the possible blockchain operations and how they are processed.

Reviewer: this link is wrong, sorry this is my fault, I think it should be (please check before applying):
```suggestion
This mechanism makes Tezos a self-amending blockchain and allows any user to propose changes to the [economic protocol](../architecture#the-tezos-self-amending-protocol), which defines the possible blockchain operations and how they are processed.
```

Author: Yup, thanks for catching that.

---
title: Governance and self-amendment
last_update:
  date: 22 February 2024
---

Tezos incorporates a built-in, on-chain mechanism for proposing, selecting, testing, and activating protocol upgrades without the need to hard fork.
This mechanism makes Tezos a self-amending blockchain and allows any user to propose changes.

All of these self-amendment features, including the scheduling and operation of each amendment period and the voting processes, are automatic and handled by the protocol without human intervention.

## Amendment periods

The self-amendment process is split into 5 periods:

> - Proposal period: Users propose changes to the Tezos protocol and vote on a single proposal to explore; the top-voted proposal moves to the next period

Reviewer 2: ```suggestion
- Proposal period: bakers propose changes to the Tezos protocol, by submitting to vote and upvoting protocol amendment proposals. If a quorum is met, the top-voted proposal moves to the next period.
```
Bakers can upvote up to 20 different proposals

Author: Changed "Other delegates can then vote on the proposals during the Proposal period up to 20 times." to "Other delegates can vote for up to 20 proposals during this period."

> - Exploration Vote period: Users vote whether to consider the top-voted proposal

Reviewer 2: ```suggestion
- Exploration period: Users vote whether to consider the top-voted proposal
```
Not sure I ever read the Vote explicit on the period names (e.g. in https://tezos.gitlab.io/active/voting.html#periods) Where is this coming from?

> - Cooldown period: If the proposal gets enough votes in the Exploration Vote period, users have the length of the Cooldown period to review and test the new protocol

Reviewer 2: ```suggestion
- Cooldown period: If a proposal gets is selected in the Exploration period, a Cooldown period starts before the final election is made. The community can continue testing the new protocol proposal, and preparing their infrastructure, before the final decision is made. 
```
Reworded a bit.

The original explanation lingers from when the cooldown period actually forked a "testchain" (not a test network) on the side of Tezos mainnet, using the selected protocol. This feature was dropped, as indeed the period was too short, and we just renamed it to Cooldown.

Still, the "users has the length of the Cooldown period to review and test the new protocol" still reads weird. Tesing the protocol started well in advance, in protocol testnets which normally start w/ the Proposal period... and in periodict test networks like weeklynet.

> - Promotion Vote period: Users make a final vote on whether to apply the proposal

Reviewer 2: ```suggestion
- Promotion  period: Users make a final vote on whether to apply the proposal
```

- Adoption period: Users have the length of the Adoption period to adapt their code and infrastructure to the proposal, and at the end of the period it is activated automatically

> Each period lasts five baking cycles (40,960 blocks at 30-second intervals or roughly 14 days, 5 hours), comprising roughly 2 months and 10 days.

Reviewer 2: ```suggestion
Each period lasts five blockchain cycles (40,960 blocks at 30-second intervals or roughly 14 days, 5 hours), comprising roughly 2 months and 10 days.
```

> If any period leads to a failure, such as not enough votes to proceed to the next period, the process reverts to the Proposal period, effectively restarting the whole process.

Author: Removed this sentence because the doc says what happens at the end of each period.

---
title: Governance and self-amendment
last_update:
  date: 22 February 2024
---

Tezos incorporates a built-in, on-chain mechanism for proposing, selecting, testing, and activating protocol upgrades without the need to hard fork.

> This mechanism makes Tezos a self-amending blockchain and allows any user to propose changes.

Reviewer 3: Note that I added the "economic" qualifier to avoid any possible confusion with other protocols in Tezos: consensus protocol, peer-to-peer protocol, RPC protocol, etc.

BTW, the page <https://docs-staging-git-voting-trili-tech.vercel.app/architecture/rpc> should be renamed to any of "RPC interface", "RPC API", "RPCs", "Tezos RPCs" or such, because it is not about the RPC protocol itself but about a set of of RPC *calls*.

Author: Making that update here: https://github.com/trilitech/tezos-developer-docs/pull/341

Reviewer: thanks, approved

All of these self-amendment features, including the scheduling and operation of each amendment period and the voting processes, are automatic and handled by the protocol without human intervention.

This self-amendment process is separate from the less formal [Tezos Improvement Process](./governance/improvement-process).

## Amendment periods

The self-amendment process is split into 5 periods:

> - Proposal period: delegates propose changes to the Tezos protocol by submitting proposals and upvoting protocol amendment proposals.

Author: Updated to use numbered items.

If a quorum is met, the top-voted proposal moves to the next period.
- Exploration period: Users vote whether to consider the top-voted proposal
- Cooldown period: If a proposal is selected in the Exploration period, a Cooldown period starts before the final election is made. The community can continue testing the new protocol proposal and preparing their infrastructure, before the final decision is made.
- Promotion period: Users make a final vote on whether to apply the proposal
- Adoption period: Users adapt their code and infrastructure to the proposal, and at the end of the period it is activated automatically

> Each period lasts five blockchain cycles (40,960 blocks at 30-second intervals or roughly 14 days, 5 hours), comprising roughly 2 months and 10 days.

Reviewer 3: But I agree that the phrase can stay removed, as the rest of the page contains the necessary explanations.

In the Exploration period, delegates vote on whether to consider the top-ranked proposal from the previous Proposal period.
Delegates can vote either "Yea", "Nay", or "Pass" on that proposal.
"Pass" means to "not vote" on a proposal.
As in the Proposal period, a delegate's voting power is based on the amount of tez in its staking balance.

At the end of the Exploration period, the network counts the votes.
To pass, the proposal must meet both of these standards:

- Quorum: A quorum of all of the voting power in the system must vote either "Yea", "Nay", or "Pass."
The amount of the quorum changes dynamically based on previous votes, which allows the system to adjust to the amount of delegates that participate in voting.

- Supermajority: The total voting power of the Yea votes must be greater than 80% of the total voting power of the Yea and Nay votes combined.
Pass votes are not counted in this equation.

If the proposal meets both of those standards, it moves to the Cooldown period.
If not, it fails and a new Proposal period starts.

### 3. Cooldown period

The Cooldown period is a delay in the process that gives users time to review and test the new version of the protocol.

> Users set up test networks that use the new version of the protocol, verify that it works, see how their baking infrastructure works with it, and discuss the proposal with other users.

Author: Changed to "community."
